src/DataStructures.jl

```diff
@@ -60,6 +65,14 @@ module DataStructures
 
     export findkey
 
+    import Base.==
+    import Base.(:)
+    import Base.+
+    import Base.-
+    import Base.isequal
+    export sd_push!, ss_push!, smd_push!, poplast!
```

Comment:
Rather than all 3 `sd_push!, ss_push!, smd_push!` is there a single name we can overload for these that refers to what it returns, rather than hungarian notation for it's first argument?
Like `push_and_get_token!` or something? (is that what they are doing?)

Reply:
Done.  The new function is called `push_return_token!`.  Since it is a single function name, I unified the APIs by making the return variables `Bool,IntSemiToken`. The `Bool` return variable is redundant for SortedMultiDict because push always creates a new record for SortedMultiDict (i.e., the first return variable is always `true` for a SortedMultiDict).